tableaudocumentapi/multilookup_dict.py

```diff
@@ -64,3 +64,24 @@ def __getitem__(self, key):
             key = self._indexes['caption'][key]
 
         return dict.__getitem__(self, key)
+
+
+class PredicatedDictionary(object):
```

Comment:
I think the code is here is fine, but I'm a little fuzzy on the need for a new type -- would a regular dictionary or data structure do?

It just makes the code read a little more complicated for new contributors. Not a strong objection, mostly curiosity.